src/main/java/me/gnat008/perworldinventory/data/metadata/PWIMetaDataManager.java

```diff
@@ -0,0 +1,37 @@
+package me.gnat008.perworldinventory.data.metadata;
+
+import ch.jalu.injector.Injector;
+
+import me.gnat008.perworldinventory.PerWorldInventory;
+
+import org.bukkit.metadata.MetadataValue;
+import org.bukkit.entity.Player;
+
+import java.util.List;
+
+import javax.inject.Inject;
+
+public class PWIMetaDataManager {
+
+    @Inject PerWorldInventory plugin;
+    @Inject Injector injector;
+
+    PWIMetaDataManager() {}
+    public <T extends PWIMetaDataValueAbstract> T createMetadataValue(Class<T> clazz, Object... args) {
+        T instance = injector.<T>newInstance(clazz);
+        instance.init(args);
+        return instance;
+    }
+
+    public <T> T getFromPlayer(Player player, String key) {
```

Comment:
Yup, if I'm gonna make the manager explicitly aware of what implementations exists, that's the way to go. I was thinking in generic magic that only required adding a new implementation in a single file, and then it could be accessible trough the generic functions. In the beginning I had a bunch of more magic planned, but it broke down due to the inability to access static members of a class from a generic context. Since that didn't work, this is probably the way to go.